### ISSUE TITLE
Add template hierarchy diagram to docs

### ISSUE THREAD

**Issue by derekperkins** [spf13/hugo#1843]
A lot of times the template hierarchy can feel overwhelming, and the documentation is scattered among the various types and layouts. I feel like this would add a lot of clarity.
![image](https://cloud.githubusercontent.com/assets/3588778/12995484/01a3df42-d0e5-11e5-922c-fc7e070d72cd.png)

**Comment by derekperkins** [comment]
How does this look? I make no claims at being a designer, but I think this helps conceptualize things nicely.
![hugo template hierarchy](https://cloud.githubusercontent.com/assets/3588778/13000194/99ad4f2e-d114-11e5-811e-c16def718574.png)

Here it is as a vector file: [Hugo Template Hierarchy.pdf](https://github.com/spf13/hugo/files/127559/Hugo.Template.Hierarchy.pdf)

- Dotted boxes are templates that seem like they should exist to be internally consistent
- We could maybe add color or a tag to boxes noting if they are a node or a page
- Am I missing any templates or are there any other suggestions? @bep @spf13 

Once this looks good, I'd like to add a new page to the docs under Templates.

**Comment by derekperkins** [comment]
First bug: missing homepage arrows

**Comment by moorereason** [comment]
Under RSS Feeds > Main RSS, the first box in the Template Level should be `/themes/THEME/layouts/rss.xml` and should have a dotted outline.

This is fantastic, Derek.

**Comment by bep** [comment]
>This is fantastic, Derek.

Agree.

**Comment by rdwatters** [comment]
This is fantastic, @derekperkins. My question is whether this will change with V1.0? It sounds like the focus of V1 is the data model. 

However, this would have to be a living document no matter what route future versions take, and, again, it's awesome :smiley:.

**Comment by bep** [comment]
Thinking of it, when I'm done with the master/overlay template work for Hugo 0.16, this drawing might get crowded ...

**Comment by derekperkins** [comment]
@bep All the more reason to have something like this.

**Comment by derekperkins** [comment]
While it's a little unwieldy as a whole because of the horizontal size, I think the most common use case will be to look at one vertical slice. "my homepage template changes aren't taking effect, where should I look to see what's overriding it."

**Comment by anthonyfok** [comment]
Really awesome!  You have my utmost respect, @derekperkins!

**Comment by spf13** [comment]
Super helpful!! 
It would be much usable as separate images on some of the individual pages.

Do you have a non-pdf vector file that someone could use to improve the aesthetics a bit?

**Comment by derekperkins** [comment]
@spf13 The original is at draw.io which I'm happy to share with anyone. I just need an email address.

**Comment by bep** [comment]
This issue has been automatically marked as stale because it has not been commented on for at least four months.

The resources of the Hugo team are limited, and so we are asking for your help.

If this is a **bug** and you can still reproduce this error on the <code>master</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, and you feel that it is still valuable, please open a proposal at https://discuss.gohugo.io/.

This issue will automatically be closed in four months if no further activity occurs. Thank you for all your contributions.

**Comment by bep** [comment]
**Note/Update:** This issue is marked as stale, and I may have said something earlier about "opening a thread on the discussion forum". Please don't.

If this is a **bug** and you can still reproduce this error on the latest <code>release</code> or the <code>master</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, and you feel that it is still relevant and valuable, please tell us why.

**Comment by robsonsobral** [comment]
The template hierarchy is still hard to follow. The image can help.

**Comment by rdwatters** [comment]
@derekperkins do you happen to have the AI/SVG/etc design file for this? If so, would you mind sending it? This will also need to be updated thanks to @bep 's awesome improvements to the lookup, the addition of kinds/_index.md, etc.

**Comment by derekperkins** [comment]
@rdwatters I originally made this on draw.io. Here's a link to that original doc:
https://drive.google.com/file/d/0B_RrOWTJYg1bYk1YaVRGUFliSTQ/view?usp=sharing

The PDF I linked to above ([Hugo Template Hierarchy.pdf](https://github.com/spf13/hugo/files/127559/Hugo.Template.Hierarchy.pdf)) was a vectorized export from draw.io.

**Comment by rdwatters** [comment]
Excellent. Thanks much, @derekperkins

**Comment by vassudanagunta** [comment]
Just saw this for the first time. Will the diagram make it into the docs? 

The template hierarchy can indeed feel overwhelming, which in turn can makes debugging why one's design isn't rendering the way one expects. It's the primary reason I recommend a `hugo server --debug` mode in issue #3514. When used, the template and partials actually used would be listed in a "debug info" block at the bottom of the page. Perhaps it could even list the each template that it looked for before setting on the one it used.
